### DISCUSSION TITLE
about trailingSlash problem

### DISCUSSION THREAD

**Discussion by zzzhangqi** [facebook/docusaurus#10791] [Q&A]
### Environment
- Docusaurus: 3.6.0
- Node: v20.0.0
- Yarn: 1.22.17

### Issue

When I set `trailingSlash: false` and use yarn start to access `/docs`, everything works fine initially. However, when I refresh the browser on the `/docs` page, a trailing slash (/) is automatically added.

This behavior does not occur on other pages; it only happens on `/docs`.

In my `introduction.md` file, I added the following configuration: `slug: /`.

**Comment by slorber** [comment]
`/docs/` is a `routeBasePath`, creating a "namespace" for all the routes of a given plugin, similar to `baseUrl`. `trailingSlash: false` will have no effect on it, the way to think about it is kind of a "folder" where you put things in (and those things won't have a trailing slash)

**Comment by zzzhangqi** [comment]
@slorber Thank you for your reply, but I'm still not quite sure how to configure it. I want to achieve the same effect as `https://docusaurus.io/docs`, where the trailing slash `/` is not added even when refreshing the page. How should I configure it?

Thank you!

**Reply by slorber** [reply]
Sorry what I said previously was wrong, it's definitively possible to have `/docs`: we have it on our website, hosted on Netlify, or through `docusaurus serve`, and the canonical URL (SEO) doesn't contain a trailing slash.

However I disagree with you, when running `yarn start` our website doesn't have a trailing slash on `http://localhost:3000/docs` and refreshing the page doesn't add it. If you have that behavior, you'll have to provide a repro using docusaurus.new

If you only see this behavior on your production host, it may be normal, they all behave differently. Refer to these docs to understand why and learn how to configure your host provider accordingly:
- https://github.com/slorber/trailing-slash-guide
- https://bjornlu.com/blog/trailing-slash-for-frameworks
